AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLogging.kt

```diff
@@ -39,41 +47,25 @@ object CrashLogging {
                 setDebug(dataProvider.enableCrashLoggingLogs)
                 setTag("locale", dataProvider.locale?.language ?: "unknown")
                 beforeSend = SentryOptions.BeforeSendCallback { event, _ ->
-                    return@BeforeSendCallback if (dataProvider.userHasOptedOut) {
+                    return@BeforeSendCallback if (userOptOut) {
                         null
                     } else {
-                        event
+                        event.apply {
+                            user = dataProvider.userProvider()?.toSentryUser()
+                        }
                     }
                 }
             }
         }
-        setNeedsDataRefresh()
-    }
-
-    @JvmStatic
-    fun setNeedsDataRefresh() {
```

Comment:
This is one of the key points of this PR. I'd prefer to model `Tracks` API in a way that client is not responsible to force-refresh data. It should just provide data or update them if somethings changed.